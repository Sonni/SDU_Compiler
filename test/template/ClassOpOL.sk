type class1 = class {
    var v1: int;
    
    func __plus(other: class1): int
        return v1 + other->v1;
    end __plus

    func __times(other: class1): int
        return v1 * other->v1;
    end __times

    func __minus(other: class1): int
        return v1 - other->v1;
    end __minus

    func __div(other: class1): int
        return v1 / other->v1;
    end __div

    func __equal(other: class1): bool
        return v1 == other->v1;
    end __equal

    func __nequal(other: class1): bool
        return v1 != other->v1;
    end __nequal

    func __less(other: class1): bool
        return v1 < other->v1;
    end __less

    func __greater(other: class1): bool
        return v1 > other->v1;
    end __greater

    func __greaterEqual(other: class1): bool
        return v1 >= other->v1;
    end __greaterEqual

    func __lessEqual(other: class1): bool
        return v1 <= other->v1;
    end __lessEqual
};

var obj1: class1;
var obj2: class1;
var res: int;
var res2: bool;

allocate obj2;
allocate obj1;

obj1->v1 = 3;
obj2->v1 = 109;

res = obj1 + obj2;
write res;

res = obj1 * obj2;
write res;

res = obj2 - obj1;
write res;

res = obj2 / obj1;
write res;

res2 = obj1 == obj2;
write res2;

res2 = obj1 != obj2;
write res2;

res2 = obj1 < obj2;
write res2;

res2 = obj1 > obj2;
write res2;

res2 = obj1 >= obj2;
write res2;

res2 = obj1 <= obj2;
write res2;

