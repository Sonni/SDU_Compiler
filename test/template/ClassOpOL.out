112
327
106
36
False
True
True
False
False
True
